staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -207,8 +206,16 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			event.Object = &unknown
 
 			// the internal event will be versioned by the encoder
+			// BEFORE MERGE, who would ever serialize this differently?  clients don't work
+			outEvent := &metav1.WatchEvent{}
 			*internalEvent = metav1.InternalEvent(event)
-			if err := e.Encode(internalEvent); err != nil {
```

Comment:
@smarterclayton why do we use this internalEvent instead of creating the external event we know we'll have to serialize to?